### PR TITLE
Do not separate stripped and non-stripped room info 

### PR DESCRIPTION
**Pull request by zecakeh**
Since they use the same type, it is not necessary to separate them. It is not even visible on public APIs that they are separated.

- [x] Public API changes documented in changelogs (optional)

Based on #2038.

Signed-off-by: Kévin Commaille <zecakeh@tedomum.fr>

### CODE REVIEW

File: crates/matrix-sdk-sqlite/migrations/state_store/002_a_create_new_room_info.sql
Comment (Hywan) [inline crates/matrix-sdk-sqlite/migrations/state_store/002_a_create_new_room_info.sql:6]: EOL is missing.

File: crates/matrix-sdk-sqlite/migrations/state_store/002_b_replace_room_info.sql
Comment (Hywan) [inline crates/matrix-sdk-sqlite/migrations/state_store/002_b_replace_room_info.sql:6]: EOL is missing.

File: crates/matrix-sdk-base/src/rooms/normal.rs
Comment (Hywan) [inline crates/matrix-sdk-base/src/rooms/normal.rs:852]:
```rust
/// The room is in an invited state.
```

File: crates/matrix-sdk-base/src/rooms/normal.rs
Comment (Hywan) [inline crates/matrix-sdk-base/src/rooms/normal.rs:854]:
```rust
/// The room is in a left state.
```

File: crates/matrix-sdk-base/src/rooms/normal.rs
Comment (Hywan) [inline crates/matrix-sdk-base/src/rooms/normal.rs:875]: Is it possible to be in a joined + left state at the same time? Or invited + joined? I reckon those states are all mutually exclusive, right?

**Review by Hywan — APPROVED** [review]
I like the patch very much, thanks for working on this!

Some parts are updating `matrix-sdk-base/src/sliding_sync.rs` which creates a small conflicts with #2039, #2041, and #2043 (cc @andybalaam). A little bit of coordination might be necessary here I guess?

**Comment by zecakeh** [comment]
I'm going to wait for #2044 to be merged to update this. CI seems to be stuck there.

File: crates/matrix-sdk-base/src/rooms/normal.rs
Comment (zecakeh) [reply crates/matrix-sdk-base/src/rooms/normal.rs:875]: This is for filtering the rooms, so it is valid to request rooms that are joined and room that are invited.

One of the possible use case is for stores that can be be more performant without using the bitflag filter, e.g. in SQLite it's more efficient to request rooms with `room_state`  equal to `RoomState::Invited` or `RoomState::Joined` than to request all rooms and then filter them.

**Review by jplatte — APPROVED** [review]

**Comment by jplatte** [comment]
I see you added a merge commit, and GitHub is disallowing me from doing a rebase-merge. Do you want to rebase / squash before we merge this?

**Comment by zecakeh** [comment]
Alright I rebased on main.

**Comment by jplatte** [comment]
To be clear, it was just a regular question whether you wanted to, not a (strong) suggestion 😅

**Comment by zecakeh** [comment]
I prefer it that way too. :wink:

I just know that while the PR is in review the rule is to not rebase stuff.

**Pull request merged**
